### PR TITLE
mpcomputerstatus: create initial windows defender wmi collector

### PR DESCRIPTION
**Pull request by rismoney**
I am new to golang, but I figured I'd take the initiative and attempt to create a collector based on a need to get defender statistics into prometheus as noted here - https://github.com/prometheus-community/windows_exporter/issues/858

I am not able to compile this yet.  Below L84-L86 I added as I thought I needed to adjust for the proper namespace.
Any advice on how to make this work would be great!

Thank you.

### CODE REVIEW

**Comment by rismoney** [comment]
so I wonder if the right way to handle the version numbers of the components is like this:
https://medium.com/teamzerolabs/staying-on-track-of-aging-node-packages-with-monitoring-in-prometheus-f915540e72df

or

https://www.robustperception.io/exposing-the-software-version-to-prometheus

Not sure how to work this in...

File: collector/msft_mpcomputerstatus.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	AntispywareSignatureAge *prometheus.Desc
+	AntivirusSignatureAge   *prometheus.Desc
+	ComputerState           *prometheus.Desc
+	FullScanAge             *prometheus.Desc
+	NISSignatureAge         *prometheus.Desc
+	QuickScanAge            *prometheus.Desc
+}
+
+func newMSFT_MpComputerStatusCollector() (Collector, error) {
+	const subsystem = "msft_mpcomputerstatus"
```
[the remark this answers is not in the document]
Comment (carlpett) [reply collector/msft_mpcomputerstatus.go:24]: I think we'd actually bundle them together under `defender` anyway in that case, so let's go with that

**Comment by carlpett** [comment]
Hey, sorry for delays in reviews.
On the sample metrics, I think they look rather suspicious - all zeros or int_max. So probably we're not getting the data out correctly. What output do you get from powershell on that class?

As for the true/false, then I agree. Typically we'd use a gauge for each of them, so eg `windows_defender_antivirus_enabled`.

**Comment by rismoney** [comment]
Delayed here too, due to other constraints. No worries.
The numbers are actually correct despite looking suspicious, because I was testing on a well updated endpoint, that never ran a full scan.  I am going to double check those and see that they exactly reflect powershell, but as of now they are spot on.

```
Get-MpComputerStatus |select *age,computerstate


AntispywareSignatureAge : 0
AntivirusSignatureAge   : 0
FullScanAge             : 4294967295
NISSignatureAge         : 0
QuickScanAge            : 0
computerstate           : 0

```

So what I am not sure is how to properly add the code to handle booleans such as
```powershell
AntivirusEnabled          : True
```

**Comment by breed808** [comment]
What value does the WMI class return for `AntivirusEnabled`? Depending on the return value type (string or uint), you could coerce that to a 0 for a false value and a 1 for a true value.

**Comment by lenglet-k** [comment]
Hi, what's the status for this feature ?

**Comment by rismoney** [comment]
I have moved on... while this exporter is extremely performant on some things due to great optimizations, it is difficult to customize w/o recompiling for missing capabilities (ie in the telegraf win_wmi input) plugin.  Ideally a lot of the collectors should not be hardcoded around what metrics are available, but instead allow for a more user configuration driven solution.  It's a tough balance.

**Pull request closed without merging**